lib/medtronic/medtronicSimulator.js

```diff
@@ -66,6 +77,111 @@ exports.make = function(config){
   }
 
   return {
+    basal: function(event) {
+      ensureTimestamp(event);
+
+      if (currBasal != null) {
+
+        if(currBasal.deliveryType === 'temp' && event.deliveryType === 'temp' && event.duration !== 0) {
+          // temp basal was updated
+          currBasal.with_expectedDuration(currBasal.duration)
+                   .with_duration(Date.parse(event.time) - Date.parse(currBasal.time));
+          event.suppressed = _.clone(currBasal.suppressed);
```

Comment:
OK, this was news to me and may require some thought re: suppressed. Not something we've actually considered before.

In your second comment you seem to contradict yourself. In the second second, do you mean 'If you press edit while a *percent* basal is running'...?